emails/template-utils.js

```diff
@@ -0,0 +1,1068 @@
+const { ASSETS_URL } = require('./env')
+
+function reqParams(fnName, values) {
+  Object.entries(values).forEach(([name, value]) => {
+    if (value === undefined) {
+      throw new Error(`${fnName}: ${name} is not defined`)
+    }
+  })
+}
+
+function style(styles) {
+  return styles
+    .replace(/\n/g, '')
+    .replace(/  /g, '')
+    .replace(/: /g, ':')
+}
+
+function attrs(attributes) {
+  return Object.entries(attributes).reduce(
+    (htmlAttrs, [name, value]) =>
+      value === null ? htmlAttrs : `${htmlAttrs} ${name}="${value}"`,
+    ''
+  )
+}
+
+function vspace(height) {
+  return table(
+    { height, style: `height: ${height}px` },
+    `<tr><td style="height:${height}px">&nbsp;</td></tr>`
+  )
+}
+
+function asset(path) {
+  return `${ASSETS_URL}/${path}`
+}
+
+function table(attributes, content) {
+  return `
+    <table ${attrs({
+      border: '0',
+      cellpadding: '0',
+      cellspacing: '0',
+      role: 'presentation',
+      width: '100%',
+      ...attributes,
+    })}>
+      <tbody>
+        ${content}
+      </tbody>
+    </table>
+  `
+}
+
+function base({ title, subtitle }, content) {
+  return `
+    <!doctype html>
+    <html
+      lang="en"
+      xmlns="http://www.w3.org/1999/xhtml"
+      xmlns:v="urn:schemas-microsoft-com:vml"
+      xmlns:o="urn:schemas-microsoft-com:office:office"
+    >
+      <head>
+        <title>Aragon Court ${title}</title>
+        <!--[if !mso]><!-- -->
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <!--<![endif]-->
+        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <style type="text/css">
+          #outlook a {
+            padding: 0;
+          }
+          body {
+            width: 100% !important;
+            height: 100%;
+            margin: 0;
+            padding: 0;
+            -webkit-text-size-adjust: 100%;
+            -ms-text-size-adjust: 100%;
+          }
+          table,
+          td {
+            border-collapse: collapse;
+            mso-table-lspace: 0pt;
+            mso-table-rspace: 0pt;
+          }
+          img {
+            border: 0;
+            height: auto;
+            line-height: 100%;
+            outline: none;
+            text-decoration: none;
+            -ms-interpolation-mode: bicubic;
+          }
+          p {
+            display: block;
+            margin: 13px 0;
+          }
+        </style>
+        <!--[if mso]>
+          <xml>
+          <o:OfficeDocumentSettings>
+            <o:AllowPNG/>
+            <o:PixelsPerInch>96</o:PixelsPerInch>
+          </o:OfficeDocumentSettings>
+          </xml>
```

Comment:
On `base2`, I was seeing litmus not show images on office / outlook clients until I added this to the `<head>`

Reply:
Ok nevermind, I lied; it's the same result with and without this bit of xml. Windows 10 mail just doesn't want to load the image.

Reply:
Ah yes I need to check why this was generated by MJML :thinking: